### PR TITLE
Docker: Build without example (leaves mounted directories unchanged)

### PR DESCRIPTION
**Pull request by ericdill**
attn @jakirkham

Attempting to resolve an issue raised in this comment thread:
https://github.com/conda-forge/staged-recipes/pull/1194/files/e77aa1bae3df32f9ed70118e26fe7ae159214856#r73998053

### CODE REVIEW

File: scripts/run_docker_build.sh [hunk 1]
```diff
@@ -16,13 +16,20 @@ show_channel_urls: true
...
+mv ${REPO_ROOT}/recipes/example $TEMP_EXAMPLES_LOCATION
 
 cat << EOF | docker run -i \
                         -v ${REPO_ROOT}/recipes:/conda-recipes \
                         -a stdin -a stdout -a stderr \
                         $IMAGE_NAME \
                         bash || exit $?
-
+echo "Will attempt to build recipes in the following directories"
+ls /conda-recipes
```
Comment (jakirkham) [inline scripts/run_docker_build.sh]: `conda-build-all` already tells one this IIRC.

**Comment by jakirkham** [comment]
Nice solution, @ericdill. Seem reasonable to me. Would remove the one line above otherwise would be happy to merge.

File: scripts/run_docker_build.sh (hunk 1 above)
Comment (ericdill) [inline scripts/run_docker_build.sh]: Oh yeah that's right. :+1:

**Comment by ericdill** [comment]
Thanks @jakirkham . Removed that unnecessary debug output.

**Comment by jakirkham** [comment]
Actually one more thing, we seem to be doing this outside the docker container. Is there any reason not to do this in the docker container?

**Comment by ericdill** [comment]
If you did this inside the docker container then you would need to do the moving, then call conda-build-all, then move it back to the /conda-recipes folder.  If you kill the docker container:
```bash
docker kill `docker ps -q`
```
then the example folder will still be removed.  I think you're right that this should be done inside the docker container, but I'd prefer to make a copy of the `/conda-recipes` folder which is the read/write copy of the host file system into a new location inside the docker container that does not affect the host system at all.  Bonus is that it is a simpler solution too
```bash
# inside docker container
cp -r /conda-recipes /conda-recipes-without-example 
rm -r /conda-recipes-without-example/example
conda-build-all /conda-recipes-without-example ...
```

**Comment by jakirkham** [comment]
Seems like that could be resolved with a [`trap`]( http://tldp.org/LDP/Bash-Beginners-Guide/html/sect_12_02.html ). Though I don't know if `docker kill` will respect that.

Honestly I've never loved that the way we do things requires us to `docker kill`. Would love if we could just do `^c` and still have it work on CircleCI. The TTY constraint bites us though. At least, I think it does. Haven't actually tried it.

**Comment by jakirkham** [comment]
Kind of an interesting read on [signals and Docker]( https://medium.com/@gchudnov/trapping-signals-in-docker-containers-7a57fdda7d86#.7y0jvtv7k ). Would have to see what `tini` does in this case.

**Comment by jakirkham** [comment]
FWICT I think tini *should* be able to do this. In any event, this totally an aside from the main issue.

**Comment by ericdill** [comment]
> Honestly I've never loved that the way we do things requires us to `docker kill`

This does not seem like a problem that is unique to the conda-forge workflow.  Well, let me hedge that  a bit.  I have not figured out a way to invoke docker containers that does not involve executing docker kill at some point to end their execution, and so I've just gotten used to ending container execution that way.  Do you find yourself also running docker kill pretty regularly to prematurely end your containers?

**Comment by ericdill** [comment]
ugh forgot to [skip ci] and [skip appveyor] yet again...

**Comment by jakirkham** [comment]
I like this solution much more, @ericdill. This is very nice. 👍

**Comment by jakirkham** [comment]
> Do you find yourself also running docker kill pretty regularly to prematurely end your containers?

I run with a TTY normally. So `docker run --rm -it condaforge/linux-anvil`. Then I can use it basically like a normal terminal. Then use `^d` to exit. Can throw in some `-v` and `-w` for various mounts and starting points.

**Comment by jakirkham** [comment]
Also seems to work fine in the CI. LGTM. Thanks @ericdill.

**Comment by ericdill** [comment]
> I run with a TTY normally. So docker run --rm -it condaforge/linux-anvil. Then I can use it basically like a normal terminal. Then use ^d to exit. Can throw in some -v and -w for various mounts and starting points.

Ah ok.  That's what I'm doing now to try and sort out some really strange python 3.5 errors.

File: scripts/run_docker_build.sh [hunk 2]
```diff
@@ -23,6 +23,10 @@ cat << EOF | docker run -i \
                         $IMAGE_NAME \
                         bash || exit $?
 
+# Copy the host recipes folder so we don't ever muck with it
+cp -r /conda-recipes /conda-recipes-without-example
+rm -r /conda-recipes-without-example/example
```
Comment (jakirkham) [inline scripts/run_docker_build.sh]: Minor comment. Can we please make this `-rf`.

Comment (ericdill) [inline scripts/run_docker_build.sh]: Done.

**Comment by jakirkham** [comment]
Thanks again @ericdill.

File: scripts/run_docker_build.sh [hunk 3]
```diff
@@ -40,14 +44,11 @@ conda install conda-build-all
 conda install conda-forge-build-setup
 source run_conda_forge_build_setup
 
-# We don't need to build the example recipe.
-rm -rf /conda-recipes/example
-
 # yum installs anything from a "yum_requirements.txt" file that isn't a blank line or comment.
 find conda-recipes -mindepth 2 -maxdepth 2 -type f -name "yum_requirements.txt" \
```
Comment (jakirkham) [inline scripts/run_docker_build.sh]: Just now realizing this, we should have renamed this one to `conda-recipes-without-example` too. However, it didn't matter much as the `example` didn't have `yum_requirements.txt`. Figured I'd mention it though.

**Pull request merged**